fapolicy_analyzer/ui/main_window.py

```diff
@@ -381,29 +383,12 @@ def on_rulesAdminMenu_activate(self, *args, **kwargs):
         self.__set_trustDbMenu_sensitive(True)
 
     def on_profileExecMenu_activate(self, *args):
-        logging.debug("MainWindow::on_profileExecMenu_activate()")
-        dlgProfTest = ProfileDialog()
-        response = dlgProfTest.get_ref().run()
-
-        if response == Gtk.ResponseType.OK:
-            profiling_args = dlgProfTest.get_text()
-            logging.debug(f"Entry text = {profiling_args}")
-            self._fapd_profiler.start_prof_session(profiling_args)
-            fapd_prof_stderr = self._fapd_profiler.fapd_prof_stderr
-            logging.debug(f"Started prof session, stderr={fapd_prof_stderr}")
-
-            # Catch termination event here somehow. Monitor pid? Pass in
-            # object returned from start_prof_session() above, or maintain only
-            # in the faprofiler instance.
-            sleep(4)
-            self._fapd_profiler.stop_prof_session()
-
-        # Add termination and/or close button
-        dlgProfTest.get_ref().destroy()
-
-        self.__pack_main_content(
-            router(ANALYZER_SELECTION.ANALYZE_FROM_AUDIT, fapd_prof_stderr)
-        )
+        page = router(ANALYZER_SELECTION.PROFILER)
+        width = self.get_object("mainWindow").get_size()[0]
+        page.get_object("dirEntry").set_property("width_request", int(width * Sizing.PROFILER_TEXT_ENTRY))
+        page._fapd_mgr = self._fapd_mgr
+        page._fapd_profiler = self._fapd_profiler
```

Comment:
These properties should be set with public setters instead of access the "protected" class properties. The other option would be to accept these as params to the `__init__` of `ProfilerPage`. The `router` gives the option of passing additional data to the `__inti__` method of the route. That is probably a cleaner way to handle this.

Reply:
Also, the `self._fapd_profiler` is created here and only used to pass into the profiler page. I think you can just move its creation right into the `__init__` of `ProfilerPage` and eliminate it as a parameter that is passed in.

Reply:
I didn't want to make a second instance of the _fapd_manager inside of main_window, but I agree that passing it in through the router is a cleaner option

Reply:
Yeah I agree about not making a 2nd copy of `_fapd_manager`, but `_fapd_profiler` is only used in `main_window` to pass it thru to the profiler page. Why not just instantiate the `_fapd_profiler` there?